packages/pigeon/lib/java_generator.dart

```diff
@@ -504,6 +568,71 @@ class JavaGenerator extends StructuredGenerator<JavaOptions> {
     indent.newln();
   }
 
+  void _writeCodecOverflowUtilities(
+    JavaOptions generatorOptions,
+    Root root,
+    Indent indent,
+    List<EnumeratedType> types, {
+    required String dartPackageName,
+  }) {
+    final NamedType overflowInteration = NamedType(
+        name: 'type',
+        type: const TypeDeclaration(baseName: _forceInt, isNullable: false));
+    final NamedType overflowObject = NamedType(
+        name: 'wrapped',
+        type: const TypeDeclaration(baseName: 'Object', isNullable: true));
+    final List<NamedType> overflowFields = <NamedType>[
+      overflowInteration,
+      overflowObject,
+    ];
+    final Class overflowClass =
+        Class(name: _overflowClassName, fields: overflowFields);
+
+    _writeDataClassSignature(generatorOptions, indent, overflowClass, () {
+      writeClassEncode(
+        generatorOptions,
+        root,
+        indent,
+        overflowClass,
+        dartPackageName: dartPackageName,
+      );
+
+      indent.format('''
+static @Nullable Object fromList(@NonNull ArrayList<Object> ${varNamePrefix}list) {
+  $_overflowClassName wrapper = new $_overflowClassName();
+  Object type = ${varNamePrefix}list.get(0);
```

Comment:
Couldn't these two locals just be inlined? They don't seem to add much here.

Reply:
If I changed the class signature to not be shared with data classes.